In [1]:
import sys
print(sys.executable)

C:\Users\kirill\AppData\Local\Programs\Python\Python313\python.exe


In [2]:
import pandas as pd

In [5]:
df = pd.read_csv("../../data/entidades_CV.csv")

In [6]:
df.head()

,WKT,id,ni_centro,tipo_centro,nombre,max,entidad,domici,leyenda,cod_ine_mun,...,comarca,municipio,sector,clasificacion,pobtotal,email,telefono,f_resolu,x_25830,y_25830
0,POINT (698055 4211247),1,29.0,CENTROS SOCIALES,"CENTRO CIVICO-SOCIAL ""VIRGEN DEL PILAR""",0.0,AYUNTAMIENTO DE LOS MONTESINOS,"PZA.DE LA IGLESIA, 2",Entidad Local,3903,...,el Baix Segura/La Vega Baja,Los Montesinos,POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE ...,POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE ...,4912.0,zero@gva.es,966721087,1995-09-12 00:00:00,698055.0,4211247.0
1,POINT (694195 4205729),2,31.0,CENTROS SOCIALES,CENTRO SOCIAL DE SAN MIGUEL DE SALINAS,0.0,AYUNTAMIENTO DE SAN MIGUEL DE SALINAS,"VICENTE BLASCO IBAÑEZ, 1",Entidad Local,3120,...,el Baix Segura/La Vega Baja,San Miguel de Salinas,POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE ...,POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE ...,6193.0,zero@gva.es,965720001,1995-09-12 00:00:00,694195.0,4205729.0
2,POINT (717780 4247138),3,32.0,HOGARES Y CLUBS PARA PERSONAS MAYORES,CLUB DE CONVIVENCIA DE TERCERA EDAD EL CISNE,0.0,AJUNTAMENT D'ALACANT,"CISNE, 3",Entidad Local,3014,...,l'Alacantí,Alacant/Alicante,PERSONAS MAYORES,PERSONAS MAYORES,330525.0,zero@gva.es,965287678,1995-09-12 00:00:00,717780.0,4247138.0
3,POINT (726984 4373714),4,36.0,RESIDENCIAS PARA PERSONAS MAYORES DEPENDIENTES,RESIDENCIA P.M.D. NTRA.SRA.DEL PUIG,31.0,"AMPARO CAMPS, S.L.","CUENCA TRAMOYERES, 5-1",Entidad Privada,46250,...,València,València,PERSONAS MAYORES,PERSONAS MAYORES,790201.0,resipuig@resipuig.com,963601081,1995-06-26 00:00:00,726984.0,4373714.0
4,POINT (700887 4236354),5,43.0,CENTROS SOCIALES,CENTRO SOCIAL PLA SECTOR V (EN PROCESO DE REGU...,0.0,AJUNTAMENT D'ELX,"AVDA. LLAURADORS, 1",Entidad Local,3065,...,el Baix Vinalopó,Elx/Elche,POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE ...,POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE ...,227659.0,zero@gva.es,965437927,1996-03-27 00:00:00,700887.0,4236354.0


In [5]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1697 entries, 0 to 1696
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   WKT            1697 non-null   str    
 1   id             1697 non-null   int64  
 2   ni_centro      1665 non-null   float64
 3   tipo_centro    1697 non-null   str    
 4   nombre         1697 non-null   str    
 5   max            1665 non-null   float64
 6   entidad        1697 non-null   str    
 7   domici         1689 non-null   str    
 8   leyenda        1697 non-null   str    
 9   cod_ine_mun    1697 non-null   int64  
 10  provincia      1697 non-null   str    
 11  comarca        1697 non-null   str    
 12  municipio      1697 non-null   str    
 13  sector         1697 non-null   str    
 14  clasificacion  1697 non-null   str    
 15  pobtotal       1665 non-null   float64
 16  email          1531 non-null   str    
 17  telefono       1688 non-null   str    
 18  f_resolu       1574

In [6]:
df.isna().sum()

WKT                0
id                 0
ni_centro         32
tipo_centro        0
nombre             0
max               32
entidad            0
domici             8
leyenda            0
cod_ine_mun        0
provincia          0
comarca            0
municipio          0
sector             0
clasificacion      0
pobtotal          32
email            166
telefono           9
f_resolu         123
x_25830            0
y_25830            0
dtype: int64

In [7]:
df.nunique()

WKT              1490
id               1697
ni_centro        1645
tipo_centro        64
nombre           1660
max               147
entidad           725
domici           1576
leyenda             9
cod_ine_mun       333
provincia           3
comarca            33
municipio         333
sector              7
clasificacion       6
pobtotal          331
email             873
telefono         1198
f_resolu         1025
x_25830          1467
y_25830          1472
dtype: int64

In [8]:
df["id"].is_unique


True

In [9]:
df[["ni_centro", "max", "pobtotal", "x_25830", "y_25830"]].describe()

,ni_centro,max,pobtotal,x_25830,y_25830
count,1665.000000,1665.000000,1665.000000,1697.000000,1.697000e+03
mean,4362.221021,35.167568,158741.714114,724180.646783,4.344673e+06
std,3824.084120,41.902705,262174.336004,22814.508146,6.922154e+04
min,29.000000,0.000000,46.000000,640597.000000,4.193060e+06
25%,1491.000000,4.000000,10537.000000,713051.000000,4.288594e+06
50%,3107.000000,20.000000,31246.000000,723355.000000,4.367918e+06
75%,4079.000000,52.000000,170990.000000,737451.000000,4.383088e+06
max,11351.000000,356.000000,790201.000000,794770.000000,4.512123e+06


In [10]:
(df["ni_centro"].dropna() % 1 == 0).all()

np.True_

In [11]:
for col in ["max", "pobtotal"]:
    print(col, (df[col].dropna() % 1 == 0).all())

max True
pobtotal True


In [12]:
df["provincia"].value_counts()

provincia
València/Valencia     884
Alacant/Alicante      489
Castelló/Castellón    324
Name: count, dtype: int64

## Initial schema observations

- The dataset contains 1697 rows and 21 columns.
- `id` has 1697 unique, non-null values, so it is a strong candidate key.
- Missing values were found in:
  - `ni_centro`: 32
  - `max`: 32
  - `domici`: 8
  - `pobtotal`: 32
  - `email`: 166
  - `telefono`: 9
  - `f_resolu`: 123
- `ni_centro`, `max`, and `pobtotal` are stored by pandas as `float64`, but all non-null values are integers.
  - Probable logical type: nullable INTEGER.
- `provincia` has no missing values and only 3 distinct values:
  - València/Valencia: 884
  - Alacant/Alicante: 489
  - Castelló/Castellón: 324
- `provincia` therefore behaves like a categorical field.

In [13]:
for col in ["sector", "clasificacion", "leyenda", "tipo_centro"]:
    print("\n", col)
    print(df[col].value_counts())


 sector
sector
PERSONAS MAYORES                                                743
PERSONAS CON DISCAPACIDAD                                       419
PERSONAS CON ENFERMEDAD MENTAL                                  244
POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE DESFAVORECIDOS    243
PERSONAS MIGRANTES                                               41
INFANCIA                                                          5
MUJERES                                                           2
Name: count, dtype: int64

 clasificacion
clasificacion
PERSONAS MAYORES                                                736
PERSONAS CON DISCAPACIDAD                                       413
PERSONAS CON ENFERMEDAD MENTAL                                  239
POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE DESFAVORECIDOS    236
PERSONAS MIGRANTES                                               41
SIN RESOLUCIÓN                                                   32
Name: count, dtype: int64

 leyenda
leyenda


### Categorical fields

- `sector` contains 7 distinct categories.
- `clasificacion` contains 6 distinct categories.
- `leyenda` contains 9 distinct categories.
- `tipo_centro` contains 64 distinct categories.

The values returned by `value_counts()` are integer frequencies, while the original column values remain strings.

`SIN RESOLUCIÓN` and `SIN INFORMAR` are not null values; they are explicit text values that may represent semantically missing information.

`sector` and `clasificacion` appear to describe related concepts, but they are not identical and should be analyzed separately.

### Campos categóricos

- `sector` contiene 7 categorías distintas.
- `clasificacion` contiene 6 categorías distintas.
- `leyenda` contiene 9 categorías distintas.
- `tipo_centro` contiene 64 categorías distintas.

Los valores devueltos por `value_counts()` son frecuencias enteras, mientras que los valores originales de las columnas siguen siendo cadenas de texto.

`SIN RESOLUCIÓN` y `SIN INFORMAR` no son valores nulos; son valores de texto explícitos que pueden representar información ausente desde el punto de vista semántico.

`sector` y `clasificacion` parecen describir conceptos relacionados, pero no son idénticos y deben analizarse por separado.

In [15]:
pd.crosstab(df["sector"], df["clasificacion"])

clasificacion,PERSONAS CON DISCAPACIDAD,PERSONAS CON ENFERMEDAD MENTAL,PERSONAS MAYORES,PERSONAS MIGRANTES,POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE DESFAVORECIDOS,SIN RESOLUCIÓN
sector,,,,,,
INFANCIA,0,0,0,0,0,5
MUJERES,0,0,0,0,0,2
PERSONAS CON DISCAPACIDAD,413,0,0,0,0,6
PERSONAS CON ENFERMEDAD MENTAL,0,239,0,0,0,5
PERSONAS MAYORES,0,0,736,0,0,7
PERSONAS MIGRANTES,0,0,0,41,0,0
POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE DESFAVORECIDOS,0,0,0,0,236,7


In [16]:
(df["sector"] == df["clasificacion"]).value_counts()

True     1665
False      32
Name: count, dtype: int64

### Relationship between `sector` and `clasificacion`

The cross-tabulation shows that `sector` and `clasificacion` are strongly related but are not identical fields.

For most records, the values are the same. The differences occur when `clasificacion` contains `SIN RESOLUCIÓN`, while `sector` still contains a specific target group.

This suggests that `sector` represents the target sector/group, whereas `clasificacion` may represent a classification status that has not yet been resolved for some records.

### Relación entre `sector` y `clasificacion`

La tabla cruzada muestra que `sector` y `clasificacion` están fuertemente relacionados, pero no son campos idénticos.

En la mayoría de los registros los valores coinciden. Las diferencias aparecen cuando `clasificacion` contiene `SIN RESOLUCIÓN`, mientras que `sector` sigue indicando un grupo objetivo específico.

Esto sugiere que `sector` representa el sector o colectivo destinatario, mientras que `clasificacion` puede representar una clasificación que todavía no se ha resuelto en algunos registros.

In [18]:
df["f_resolu"].dropna().head(10)

0    1995-09-12 00:00:00
1    1995-09-12 00:00:00
2    1995-09-12 00:00:00
3    1995-06-26 00:00:00
4    1996-03-27 00:00:00
5    1998-07-24 00:00:00
6    1998-07-24 00:00:00
7    1999-02-26 00:00:00
8    1995-09-12 00:00:00
9    1998-01-09 00:00:00
Name: f_resolu, dtype: str

In [19]:
parsed_dates = pd.to_datetime(df["f_resolu"], errors="coerce")

In [20]:
(df["f_resolu"].notna() & parsed_dates.isna()).sum()

np.int64(0)

In [21]:
df["f_resolu"].isna().sum()

np.int64(123)

In [22]:
parsed_dates.dropna().dt.time.value_counts()

f_resolu
00:00:00    1562
14:44:52       1
16:14:52       1
14:24:52       1
15:34:52       1
12:34:52       1
08:54:52       1
15:14:52       1
15:24:52       1
10:04:52       1
09:04:52       1
13:04:52       1
13:44:52       1
Name: count, dtype: int64

In [23]:
parsed_dates.min(), parsed_dates.max()

(Timestamp('1977-12-31 00:00:00'), Timestamp('2026-08-04 13:44:52'))

In [24]:
df.loc[
    parsed_dates.notna() & (parsed_dates.dt.time != pd.Timestamp("00:00:00").time()),
    ["id", "f_resolu"]
]

,id,f_resolu
1653,1654,2026-01-21 14:44:52
1654,1655,2026-01-21 16:14:52
1655,1656,2026-02-05 14:24:52
1656,1657,2026-03-11 15:34:52
1657,1658,2026-04-02 12:34:52
1658,1659,2026-04-27 08:54:52
1659,1661,2026-04-30 15:14:52
1660,1662,2026-04-30 15:24:52
1661,1663,2026-05-26 10:04:52
1662,1664,2026-07-09 09:04:52


### `f_resolu` — date/time analysis

- The dataset contains 1697 records.
- `f_resolu` has 1574 non-null values and 123 missing values.
- All 1574 non-null values can be successfully parsed as date/time values.
- No malformed non-null date/time values were detected.
- 1562 values contain the time `00:00:00`.
- 12 values contain a non-zero time component.
- The observed date/time range is from `1977-12-31 00:00:00` to `2026-08-04 13:44:52`.
- All 12 records with a non-zero time are recent records from 2026, with IDs approximately between 1654 and 1666.
- Their times vary, but all observed values have `52` seconds.
- This may indicate that newer records are stored with a real timestamp, while older records were stored only with a date and represented as midnight.
- In the source file the field is stored as `str`, but its logical type is most likely `DATETIME` / `TIMESTAMP`.
- The business meaning of the time component should be verified before converting the field to a pure `DATE`.

### `f_resolu` — análisis de fecha y hora

- El conjunto de datos contiene 1697 registros.
- `f_resolu` tiene 1574 valores no nulos y 123 valores ausentes.
- Los 1574 valores no nulos pueden convertirse correctamente a valores de fecha y hora.
- No se detectaron valores de fecha/hora incorrectos entre los valores existentes.
- 1562 valores contienen la hora `00:00:00`.
- 12 valores contienen una hora distinta de cero.
- El intervalo observado va desde `1977-12-31 00:00:00` hasta `2026-08-04 13:44:52`.
- Los 12 registros con una hora distinta de cero son registros recientes de 2026, con identificadores aproximadamente entre 1654 y 1666.
- Las horas son diferentes, pero todos los valores observados tienen `52` segundos.
- Esto puede indicar que los registros más recientes se almacenan con una marca temporal completa, mientras que los registros antiguos se almacenaban únicamente con la fecha y se representaban con la hora `00:00:00`.
- En el archivo de origen el campo está almacenado como `str`, pero su tipo lógico más probable es `DATETIME` / `TIMESTAMP`.
- Antes de convertir el campo a un simple `DATE`, debería comprobarse el significado de negocio de la parte horaria.

In [25]:
df["telefono"].dropna().head(15)

0     966721087
1     965720001
2     965287678
3     963601081
4     965437927
5     962807204
6     962807204
7     963942044
8     965855500
9     964329293
10    965587168
11    965426363
12    965583301
13    965583301
14    965583301
Name: telefono, dtype: str

In [26]:
df["telefono"].dropna().str.len()

0       9
1       9
2       9
3       9
4       9
       ..
1692    9
1693    9
1694    9
1695    9
1696    9
Name: telefono, Length: 1688, dtype: int64

In [27]:
df["telefono"].value_counts()

telefono
0            68
963393090    23
963665070    15
962965155    13
964219101    12
             ..
964547130     1
962965003     1
964220457     1
654588658     1
962101001     1
Name: count, Length: 1198, dtype: int64

In [28]:
df["telefono"].dropna().str.len().value_counts()

telefono
9     1604
1       68
10       4
12       3
8        3
3        3
20       2
19       1
Name: count, dtype: int64

In [30]:
df.loc[
    (df["telefono"].notna())
    & (df["telefono"] != "0")
    & (df["telefono"].str.len() != 9),
    ["id", "telefono"]
]

,id,telefono
659,660,96 189 90 97
803,804,964398869 - 96439828
852,853,96265155
856,857,96265155
1068,1069,964360002; 964367054
1360,1361,9658150583
1361,1362,9658150583
1362,1363,9658150583
1432,1433,96 486 79 22
1470,1471,96592161


In [31]:
only_digits = df["telefono"].str.replace(r"\D", "", regex=True)

In [32]:
digit_count = only_digits.str.len()

In [33]:
type(digit_count)

pandas.Series

In [34]:
digit_count.dtype

dtype('float64')

In [35]:
digit_count.head(10)

0    9.0
1    9.0
2    9.0
3    9.0
4    9.0
5    9.0
6    9.0
7    9.0
8    9.0
9    9.0
Name: telefono, dtype: float64

In [37]:
mask = (
    (df["telefono"].notna())
    & (df["telefono"] != "0")
    & (df["telefono"].str.len() != 9)
)

phones_check = df.loc[mask, ["id", "telefono"]].copy()

phones_check["only_digits"] = (
    phones_check["telefono"]
    .str.replace(r"\D", "", regex=True)
)

phones_check["digit_count"] = phones_check["only_digits"].str.len()

phones_check.sort_values("digit_count", ascending=False)

,id,telefono,only_digits,digit_count
1068,1069,964360002; 964367054,964360002964367054,18
1600,1601,961366865 676422994,961366865676422994,18
803,804,964398869 - 96439828,96439886996439828,17
1360,1361,9658150583,9658150583,10
1362,1363,9658150583,9658150583,10
1361,1362,9658150583,9658150583,10
1579,1580,9636301007,9636301007,10
659,660,96 189 90 97,961899097,9
1542,1543,964 51 50 14,964515014,9
1432,1433,96 486 79 22,964867922,9


### `telefono` — format and data quality analysis

- The field is stored as `str`, which is appropriate because telephone numbers are identifiers rather than numeric values used in calculations.
- There are 1688 non-null values and 9 missing values.
- Most values (1604 records) contain exactly 9 characters.
- 68 records contain the explicit value `"0"`. This is not a null value technically, but it most likely represents a missing or unknown telephone number.
- Several records use formatting characters such as spaces, hyphens, and semicolons.
- Three values with spaces become valid 9-digit strings after removing non-digit characters.
- Some records appear to contain two telephone numbers in a single field, for example:
  - `964360002; 964367054`
  - `961366865 676422994`
- Other anomalous values contain 10, 8, or 3 digits after normalization.
- Therefore, the logical type of the field is `STRING`, but the source contains both formatting inconsistencies and semantically missing values.

#### Possible normalization rules for integration

A later transformation step could use regular expressions to:

1. remove spaces and other formatting characters;
2. convert `"0"` to a null value;
3. detect multiple telephone numbers stored in the same field;
4. split multiple telephone numbers when necessary;
5. validate normalized telephone numbers against the expected length;
6. flag values whose normalized length is not 9 digits.

If the global schema supports only one telephone number, a documented rule such as selecting the first number could be applied. If multiple telephone numbers are supported, they should preferably be preserved rather than discarded.

### `telefono` — análisis del formato y de la calidad de los datos

- El campo está almacenado como `str`, lo cual es adecuado porque los números de teléfono son identificadores y no valores numéricos utilizados para cálculos.
- Hay 1688 valores no nulos y 9 valores ausentes.
- La mayoría de los valores (1604 registros) contienen exactamente 9 caracteres.
- 68 registros contienen el valor explícito `"0"`. Técnicamente no es un valor nulo, pero probablemente representa un número de teléfono ausente o desconocido.
- Algunos registros utilizan caracteres de formato como espacios, guiones y punto y coma.
- Tres valores con espacios se convierten en cadenas válidas de 9 dígitos después de eliminar los caracteres no numéricos.
- Algunos registros parecen contener dos números de teléfono en un mismo campo, por ejemplo:
  - `964360002; 964367054`
  - `961366865 676422994`
- Otros valores anómalos contienen 10, 8 o 3 dígitos después de la normalización.
- Por tanto, el tipo lógico del campo es `STRING`, pero el origen contiene tanto inconsistencias de formato como valores semánticamente ausentes.

#### Posibles reglas de normalización para la integración

En una etapa posterior de transformación se podrían utilizar expresiones regulares para:

1. eliminar espacios y otros caracteres de formato;
2. convertir `"0"` en un valor nulo;
3. detectar varios números de teléfono almacenados en un mismo campo;
4. separar varios números cuando sea necesario;
5. validar los teléfonos normalizados según la longitud esperada;
6. marcar como anomalías los valores cuya longitud normalizada no sea de 9 dígitos.

Si el esquema global solo admite un número de teléfono, se podría aplicar una regla documentada, por ejemplo seleccionar el primer número. Si el esquema permite varios teléfonos, sería preferible conservarlos en lugar de descartar información.

In [5]:
df["email"].dtype

dtype('O')

In [9]:
df["email"].isna().sum()

np.int64(166)

In [10]:
df["email"].notna().sum()

np.int64(1531)

In [11]:
df["email"].nunique()

873

In [12]:
df["email"].dropna().head(15)

0                                 zero@gva.es
1                                 zero@gva.es
2                                 zero@gva.es
3                       resipuig@resipuig.com
4                                 zero@gva.es
5                                 zero@gva.es
6                                 zero@gva.es
7     RESIDENCIAJARDINESDELPARTERRE@GMAIL.COM
8                                 zero@gva.es
9                        EL-RINCONET@YAHOO.ES
10                                zero@gva.es
11                                zero@gva.es
12                AJUMTAMENT@VALLDE LAGUAR.ES
13                AJUMTAMENT@VALLDE LAGUAR.ES
14                AJUMTAMENT@VALLDE LAGUAR.ES
Name: email, dtype: object

In [13]:
df["email"].dropna().value_counts()

email
zero@gva.es                 413
ESPURNA@ESPURNA.ORG          19
MTURRO@NOVAEDAT.ES           16
agomezc@accem.es              9
DIRECCION_GRANVIA@GVA.ES      7
                           ... 
ASESORIA@RONCALES.COM         1
LUCIGRAMA@HOTMAIL.COM         1
fj.perezgalvez@gmail.com      1
JMAM@FONTILLES.ORG            1
APADIS_2@CV.GVA.ES            1
Name: count, Length: 873, dtype: int64

In [14]:
(
    df["email"].notna()
    & (df["email"] != df["email"].str.strip())
).sum()

np.int64(3)

In [15]:
df.loc[(
    df["email"].notna()
    & (df["email"] != df["email"].str.strip())
), ["id", "email"]]

,id,email
626,627,direccion.elda@domusvi.es
627,628,direccion.elda@domusvi.es
780,781,cdmontaverner@gmail.com


In [16]:
(
    ~df["email"]
    .dropna()
    .str.fullmatch(r"^[^@\s]+@[^@\s]+\.[^@\s]+$")
).sum()

np.int64(17)

In [17]:
email_pattern = r"^[^@\s]+@[^@\s]+\.[^@\s]+$"

invalid_email_mask = (
    df["email"].notna()
    & ~df["email"].str.fullmatch(email_pattern, na=False)
)

df.loc[invalid_email_mask, ["id", "email"]]

,id,email
12,13,AJUMTAMENT@VALLDE LAGUAR.ES
13,14,AJUMTAMENT@VALLDE LAGUAR.ES
14,15,AJUMTAMENT@VALLDE LAGUAR.ES
159,160,TERCERA EDAD @TORREVIEJA.EU
200,201,C_O_JOSEALCAMI@HOTMAIL
315,316,ALICIA@OLIVA.ES MP@OLIVA.ES
436,437,ALICIA@OLIVA.ES MP@OLIVA.ES
626,627,direccion.elda@domusvi.es
627,628,direccion.elda@domusvi.es
687,688,INFO@VINAROS. ORG


In [18]:
clean_email = df["email"].str.strip().str.lower()

### `email` — profile and data quality analysis

- Source dtype: `object` (string data).
- The field contains 1531 non-null values and 166 null values.
- There are 873 distinct non-null values.
- `zero@gva.es` occurs 413 times and is likely a technical placeholder rather than a real contact address.
- Some email addresses use different letter case; case normalization may therefore be useful during integration.
- 3 values contain leading or trailing whitespace.
- 17 non-null values do not match the basic `name@domain.tld` email pattern.
- Observed anomalies include:
  - whitespace inside or around an address;
  - missing `@`;
  - incomplete domains;
  - incorrect punctuation;
  - multiple email addresses stored in one field.
- Logical type: `STRING`.
- Nullable: `YES`.

#### Possible integration transformations

- trim leading/trailing whitespace;
- normalize case;
- convert known placeholder values such as `zero@gva.es` to `NULL` if confirmed semantically;
- validate email syntax;
- detect and split multiple email addresses where appropriate;
- flag malformed addresses for review.

### `email` — perfil y análisis de calidad de datos

- Tipo de origen: `object` (datos de texto).
- El campo contiene 1531 valores no nulos y 166 valores nulos.
- Hay 873 valores distintos no nulos.
- `zero@gva.es` aparece 413 veces y probablemente es un valor técnico de sustitución y no una dirección de contacto real.
- Algunas direcciones utilizan mayúsculas y minúsculas de forma diferente; por tanto, puede ser útil normalizar el uso de mayúsculas/minúsculas durante la integración.
- 3 valores contienen espacios al principio o al final.
- 17 valores no nulos no cumplen el patrón básico `nombre@dominio.tld`.
- Entre las anomalías observadas se encuentran:
  - espacios dentro o alrededor de la dirección;
  - ausencia de `@`;
  - dominios incompletos;
  - signos de puntuación incorrectos;
  - varias direcciones de correo almacenadas en un mismo campo.
- Tipo lógico: `STRING`.
- Admite valores nulos: `SÍ`.

#### Posibles transformaciones para la integración

- eliminar espacios iniciales y finales;
- normalizar mayúsculas/minúsculas;
- convertir valores de sustitución conocidos como `zero@gva.es` a `NULL` si se confirma su significado;
- validar la sintaxis del correo electrónico;
- detectar y separar varias direcciones cuando sea necesario;
- marcar las direcciones incorrectas para su revisión.

In [19]:
df["cod_ine_mun"].head(15)

0      3903
1      3120
2      3014
3     46250
4      3065
5     46046
6     46046
7     46250
8      3031
9     12128
10     3110
11     3065
12     3137
13     3137
14     3137
Name: cod_ine_mun, dtype: int64

In [20]:
df["cod_ine_mun"].nunique()

333

In [21]:
df["cod_ine_mun"].value_counts()

cod_ine_mun
46250    230
12040    100
3014      75
46244     54
3065      52
        ... 
3053       1
46093      1
12021      1
46128      1
46212      1
Name: count, Length: 333, dtype: int64

In [22]:
df["cod_ine_mun"].astype(str).str.len().value_counts().sort_index()

cod_ine_mun
4     489
5    1208
Name: count, dtype: int64

In [23]:
cod_ine_str = df["cod_ine_mun"].astype(str).str.zfill(5)
cod_ine_str.head(15)

0     03903
1     03120
2     03014
3     46250
4     03065
5     46046
6     46046
7     46250
8     03031
9     12128
10    03110
11    03065
12    03137
13    03137
14    03137
Name: cod_ine_mun, dtype: object

In [24]:
pd.crosstab(
    cod_ine_str.str[:2],
    df["provincia"]
)

provincia,Alacant/Alicante,Castelló/Castellón,València/Valencia
cod_ine_mun,,,
03,489,0,0
12,0,324,0
46,0,0,884


In [25]:
municipios_per_code = df.groupby("cod_ine_mun")["municipio"].nunique()

In [26]:
municipios_per_code.value_counts()

municipio
1    333
Name: count, dtype: int64

### `cod_ine_mun` — municipal identifier analysis

- Source dtype: `int64`.
- The field contains 1697 non-null values and no missing values.
- There are 333 distinct codes.
- Although pandas stores the field as an integer, it represents an identifier rather than a measurable numeric value.
- 489 values contain 4 digits and 1208 values contain 5 digits.
- The 4-digit values are consistent with 5-digit INE municipal codes whose leading zero was lost when the CSV was interpreted as numeric data.
- Padding the values to 5 characters with leading zeros restores the expected representation.
- The first two digits correspond consistently to the province:
  - `03` → Alacant/Alicante
  - `12` → Castelló/Castellón
  - `46` → València/Valencia
- All 333 distinct municipal codes are associated with exactly one distinct `municipio`.
- Probable logical type: `STRING` / `CHAR(5)`, not `INTEGER`.
- Nullable: `NO`.

#### Possible integration transformation

Convert the source value to a 5-character string and restore leading zeros using zero-padding before mapping it to the global schema.

### `cod_ine_mun` — análisis del identificador municipal

- Tipo de origen: `int64`.
- El campo contiene 1697 valores no nulos y no tiene valores ausentes.
- Hay 333 códigos distintos.
- Aunque pandas almacena el campo como entero, representa un identificador y no una magnitud numérica.
- 489 valores contienen 4 dígitos y 1208 valores contienen 5 dígitos.
- Los valores de 4 dígitos son compatibles con códigos municipales INE de 5 dígitos cuyo cero inicial se perdió al interpretar el CSV como datos numéricos.
- Al completar los valores hasta 5 caracteres con ceros a la izquierda se recupera la representación esperada.
- Los dos primeros dígitos corresponden de forma consistente a la provincia:
  - `03` → Alacant/Alicante
  - `12` → Castelló/Castellón
  - `46` → València/Valencia
- Los 333 códigos municipales distintos están asociados exactamente con un único `municipio`.
- Tipo lógico probable: `STRING` / `CHAR(5)`, no `INTEGER`.
- Admite valores nulos: `NO`.

#### Posible transformación para la integración

Convertir el valor de origen en una cadena de 5 caracteres y restaurar los ceros iniciales antes de mapearlo al esquema global.

In [28]:
df["municipio"].dtype

dtype('O')

In [29]:
df["municipio"].isna().sum()

np.int64(0)

In [30]:
df["municipio"].nunique()

333

In [31]:
df["municipio"].head(15)

0            Los Montesinos
1     San Miguel de Salinas
2          Alacant/Alicante
3                  València
4                 Elx/Elche
5                      Barx
6                      Barx
7                  València
8                  Benidorm
9                 Vilafamés
10       el Ràfol d'Almúnia
11                Elx/Elche
12        la Vall de Laguar
13        la Vall de Laguar
14        la Vall de Laguar
Name: municipio, dtype: object

In [32]:
df["municipio"].value_counts().head(15)

municipio
València                230
Castelló de la Plana    100
Alacant/Alicante         75
Torrent                  54
Elx/Elche                52
Gandia                   47
Villena                  24
Aldaia                   23
Elda                     21
Orihuela                 21
Ontinyent                20
Vila-real                19
Alcoi/Alcoy              18
Xàtiva                   17
Sagunt/Sagunto           17
Name: count, dtype: int64

In [33]:
(
    df["municipio"] != df["municipio"].str.strip()
).sum()

np.int64(0)

In [34]:
codes_per_municipio = df.groupby("municipio")["cod_ine_mun"].nunique()

In [35]:
codes_per_municipio.value_counts()

cod_ine_mun
1    333
Name: count, dtype: int64

In [36]:
municipio_lengths = df["municipio"].str.len()

In [37]:
municipio_lengths.min()

np.int64(3)

In [38]:
municipio_lengths.max()

np.int64(47)

In [46]:
df.loc[(municipio_lengths == municipio_lengths.min()) | (municipio_lengths == municipio_lengths.max()),"municipio"].unique()

array(['Ibi', 'Sant Vicent del Raspeig/San Vicente del Raspeig', 'Cox',
       'Sax'], dtype=object)

In [47]:
municipio_unique = df[["municipio"]].drop_duplicates().copy()

In [48]:
municipio_unique["length"] = municipio_unique["municipio"].str.len()

In [49]:
municipio_unique.nsmallest(5, "length")

,municipio,length
215,Ibi,3
314,Cox,3
521,Sax,3
5,Barx,4
19,Xert,4


In [50]:
municipio_unique.nlargest(5, "length")

,municipio,length
296,Sant Vicent del Raspeig/San Vicente del Raspeig,47
41,les Alqueries/Alquerías del Niño Perdido,40
127,la Torre de les Maçanes/Torremanzanas,37
479,el Camp de Mirra/Campo de Mirra,31
989,Vilafranca/Villafranca del Cid,30


### `municipio` — municipality name analysis

- Source dtype: `object` (string data).
- The field contains 1697 non-null values and no missing values.
- There are 333 distinct municipality names.
- Municipality names are repeated because multiple entities may belong to the same municipality.
- No leading or trailing whitespace was detected.
- Each of the 333 distinct `municipio` values corresponds to exactly one `cod_ine_mun`.
- Conversely, each `cod_ine_mun` corresponds to exactly one `municipio`.
- Therefore, the observed relationship between `municipio` and `cod_ine_mun` is one-to-one.
- The five shortest distinct municipality names are:
  - `Ibi` — 3 characters
  - `Cox` — 3
  - `Sax` — 3
  - `Barx` — 4
  - `Xert` — 4
- The five longest distinct municipality names are:
  - `Sant Vicent del Raspeig/San Vicente del Raspeig` — 47 characters
  - `les Alqueries/Alquerías del Niño Perdido` — 40
  - `la Torre de les Maçanes/Torremanzanas` — 37
  - `el Camp de Mirra/Campo de Mirra` — 31
  - `Vilafranca/Villafranca del Cid` — 30
- Long bilingual names appear to be legitimate data rather than formatting anomalies.
- Logical type: `STRING`.
- Nullable: `NO`.

### `municipio` — análisis del nombre del municipio

- Tipo de origen: `object` (datos de texto).
- El campo contiene 1697 valores no nulos y no tiene valores ausentes.
- Hay 333 nombres de municipios distintos.
- Los nombres se repiten porque varias entidades pueden pertenecer al mismo municipio.
- No se detectaron espacios iniciales ni finales.
- Cada uno de los 333 valores distintos de `municipio` corresponde exactamente a un `cod_ine_mun`.
- A su vez, cada `cod_ine_mun` corresponde exactamente a un `municipio`.
- Por tanto, en los datos observados existe una relación uno-a-uno entre `municipio` y `cod_ine_mun`.
- Los cinco nombres distintos más cortos son:
  - `Ibi` — 3 caracteres
  - `Cox` — 3
  - `Sax` — 3
  - `Barx` — 4
  - `Xert` — 4
- Los cinco nombres distintos más largos son:
  - `Sant Vicent del Raspeig/San Vicente del Raspeig` — 47 caracteres
  - `les Alqueries/Alquerías del Niño Perdido` — 40
  - `la Torre de les Maçanes/Torremanzanas` — 37
  - `el Camp de Mirra/Campo de Mirra` — 31
  - `Vilafranca/Villafranca del Cid` — 30
- Los nombres bilingües largos parecen ser datos legítimos y no anomalías de formato.
- Tipo lógico: `STRING`.
- Admite valores nulos: `NO`.

In [51]:
df["comarca"].dtype

dtype('O')

In [53]:
df["comarca"].isna().sum()

np.int64(0)

In [54]:
df["comarca"].nunique()

33

In [55]:
df["comarca"].head(15)

0     el Baix Segura/La Vega Baja
1     el Baix Segura/La Vega Baja
2                      l'Alacantí
3                        València
4                el Baix Vinalopó
5                        la Safor
6                        la Safor
7                        València
8                 la Marina Baixa
9                   la Plana Alta
10                 la Marina Alta
11               el Baix Vinalopó
12                 la Marina Alta
13                 la Marina Alta
14                 la Marina Alta
Name: comarca, dtype: object

In [56]:
df["comarca"].value_counts()

comarca
València                               230
l'Horta Sud                            154
la Plana Alta                          136
l'Alacantí                             118
l'Horta Nord                            87
el Baix Segura/La Vega Baja             81
la Safor                                79
la Plana Baixa                          69
la Ribera Alta                          66
el Baix Vinalopó                        64
el Camp de Túria                        62
la Marina Alta                          61
el Vinalopó Mitjà/El Vinalopó Medio     51
la Vall d'Albaida                       46
la Marina Baixa                         39
l'Alcoià                                37
la Ribera Baixa                         35
el Baix Maestrat                        33
la Costera                              32
el Camp de Morvedre                     31
l’Alt Vinalopó/El Alto Vinalopó         29
El Alto Palancia                        26
El Alto Mijares                         25
La 

In [57]:
(df["comarca"] != df["comarca"].str.strip()).sum()

np.int64(0)

In [58]:
comarcas_per_municipio = df.groupby("municipio")["comarca"].nunique()
comarcas_per_municipio.value_counts()

comarca
1    333
Name: count, dtype: int64

In [59]:
provincias_per_comarca = df.groupby("comarca")["provincia"].nunique()
provincias_per_comarca.value_counts()

provincia
1    33
Name: count, dtype: int64

### `comarca` — regional classification analysis

- Source dtype: `object` (string data).
- The field contains 1697 non-null values and no missing values.
- There are 33 distinct comarca values.
- The field behaves as a categorical geographic attribute.
- No leading or trailing whitespace was detected.
- Each `municipio` is associated with exactly one `comarca` in the observed dataset.
- Each `comarca` is associated with exactly one `provincia`.
- The observed geographic hierarchy is therefore internally consistent:

  `provincia → comarca → municipio`

- No integrity anomalies were detected in these relationships.
- Logical type: `STRING` / categorical.
- Nullable: `NO`.

### `comarca` — análisis de la clasificación territorial

- Tipo de origen: `object` (datos de texto).
- El campo contiene 1697 valores no nulos y no tiene valores ausentes.
- Hay 33 valores distintos de `comarca`.
- El campo funciona como un atributo geográfico categórico.
- No se detectaron espacios iniciales ni finales.
- Cada `municipio` está asociado exactamente con una `comarca` en el conjunto de datos observado.
- Cada `comarca` está asociada exactamente con una `provincia`.
- Por tanto, la jerarquía geográfica observada es internamente consistente:

  `provincia → comarca → municipio`

- No se detectaron anomalías de integridad en estas relaciones.
- Tipo lógico: `STRING` / categórico.
- Admite valores nulos: `NO`.

In [60]:
df["domici"].dtype

dtype('O')

In [61]:
df["domici"].isna().sum()

np.int64(8)

In [62]:
df["domici"].nunique().sum()

AttributeError: 'int' object has no attribute 'sum'

In [63]:
df["domici"].nunique()

1576

In [64]:
df["domici"].head(15)

0                   PZA.DE LA IGLESIA, 2
1               VICENTE BLASCO IBAÑEZ, 1
2                               CISNE, 3
3                 CUENCA TRAMOYERES, 5-1
4                    AVDA. LLAURADORS, 1
5     URBANIZACION LA DROVA, C/TOSSAL 10
6     URBANIZACION LA DROVA, C/TOSSAL 10
7            PZA.ALFONSO EL MAGNANIMO, 1
8                            C/ GOYA S/N
9           calle Pintor Luis Prades, 13
10          SEGARIA, 12 (ZONA DEPORTIVA)
11                          TRASVASE, 32
12                      REI EN JAUME, 27
13                             MAYOR, 11
14                    PZA. SACRAMENTO, 1
Name: domici, dtype: object

In [65]:
df["domici"].value_counts().head(15)

domici
AVDA TARREGA MONTEBLANCO, 43              12
AVDA. VICTIMAS DEL TERRORISMO, 16          4
C/ PRADO DE LA VILLA, 92                   4
C/  TRINQUETE Nº. 29                       3
C/ MATEMATIC MARÇAL S/N                    3
Carretera Plá de Quart, Km 3               3
PZA. MAYOR, 1                              3
C/ PRADO DE LA VILLA, 92 . 3º              3
calle Mestre Josep Segarra, n.º 4          2
calle Estatut, nº24                        2
PCA. DE LES ESCOLES, S/N                   2
C/ EUGENIA VIÑES, 156                      2
Camino del Caminás 42                      2
DOCTOR FLEMING, S/N                        2
PROLONG. C/ FEDERICO GARCÍA LORCA, S/N     2
Name: count, dtype: int64

In [66]:
(df["domici"] != df["domici"].str.strip()).sum()

np.int64(8)

In [67]:
df["domici"].str.strip().eq("").sum()

np.int64(0)

In [68]:
(
    df["domici"].notna()
    & (df["domici"] != df["domici"].str.strip())
).sum()

np.int64(0)

In [69]:
domici_lengths = df["domici"].dropna().str.len()

In [70]:
domici_lengths.min(), domici_lengths.max()

(np.int64(6), np.int64(84))

In [71]:
domici_unique = df[["domici"]].dropna().drop_duplicates().copy()
domici_unique["length"] = domici_unique["domici"].str.len()

domici_unique.nsmallest(5, "length")
domici_unique.nlargest(5, "length")

,domici,length
1600,"RONDA NARCISO MONTURIOL Y ESTARRIOL 7-9 BAJOS,...",84
1530,"Calle Agent Tomás Catalán Simó, números 12 y 1...",79
1678,"C/MONTE DE SANTA POLA, 16 PL.01 LOCAL14 EDIF.3...",78
1584,"DE LA LIBERTAD, esq. C/MARIANO BENLLIURE, esq....",69
1650,"C/ JOAQUÍN MARÍA LÓPEZ, 37 ENTLO. (ANTES C/ LU...",69


In [72]:
with pd.option_context("display.max_colwidth", None):
    display(domici_unique.nlargest(5, "length"))

,domici,length
1600,"RONDA NARCISO MONTURIOL Y ESTARRIOL 7-9 BAJOS, PARQUE TECNOLÓGICO, PATERNA, VALENCIA",84
1530,"Calle Agent Tomás Catalán Simó, números 12 y 14, planta baja, primera y segunda",79
1678,"C/MONTE DE SANTA POLA, 16 PL.01 LOCAL14 EDIF.3 (URB. LIFE RESORT-GRAN ALACANT)",78
1584,"DE LA LIBERTAD, esq. C/MARIANO BENLLIURE, esq. C/VICENTE ALONSO PEREZ",69
1650,"C/ JOAQUÍN MARÍA LÓPEZ, 37 ENTLO. (ANTES C/ LUCIANO LÓPEZ FERRER, 13)",69


In [73]:
with pd.option_context("display.max_colwidth", None):
    display(domici_unique.nsmallest(5, "length"))

,domici,length
1483,PUERTO,6
220,"ALTA, 7",7
336,PLAZA 9,7
1481,BASCULA,7
1535,BARRACA,7


### `domici` — address field analysis

- Source dtype: `object` (string data).
- The field contains 1689 non-null values and 8 missing values.
- There are 1576 distinct non-null addresses.
- Repeated addresses are not necessarily anomalies, because several entities may be located at the same address.
- No leading or trailing whitespace was detected among non-null values.
- No non-null values become empty strings after trimming whitespace.
- Address length ranges from 6 to 84 characters.
- The longest values are detailed multi-part addresses and appear structurally plausible.
- The shortest values include highly abbreviated entries such as `PUERTO`, `ALTA, 7`, `PLAZA 9`, `BASCULA`, and `BARRACA`.
- These short values may represent valid abbreviated addresses, but they could also indicate incomplete address information and may require external validation.
- The field contains heterogeneous free-text address formats, including abbreviations such as `C/`, `AVDA.`, `PZA.`, street names, building numbers, and additional location details.
- Logical type: `STRING`.
- Nullable: `YES`.

#### Possible integration considerations

- preserve the original address text;
- trim whitespace during normalization;
- optionally standardize common street-type abbreviations;
- flag unusually short or otherwise suspicious address values for review if stricter address quality is required.

### `domici` — análisis del campo de dirección

- Tipo de origen: `object` (datos de texto).
- El campo contiene 1689 valores no nulos y 8 valores ausentes.
- Hay 1576 direcciones distintas no nulas.
- Las direcciones repetidas no son necesariamente anomalías, ya que varias entidades pueden estar ubicadas en la misma dirección.
- No se detectaron espacios iniciales ni finales entre los valores no nulos.
- Ningún valor no nulo se convierte en una cadena vacía después de eliminar los espacios.
- La longitud de las direcciones varía entre 6 y 84 caracteres.
- Los valores más largos son direcciones detalladas con varias partes y parecen estructuralmente plausibles.
- Entre los valores más cortos aparecen entradas muy abreviadas como `PUERTO`, `ALTA, 7`, `PLAZA 9`, `BASCULA` y `BARRACA`.
- Estos valores cortos pueden representar direcciones abreviadas válidas, aunque también podrían indicar información de dirección incompleta y requerir validación externa.
- El campo contiene formatos heterogéneos de dirección en texto libre, incluyendo abreviaturas como `C/`, `AVDA.`, `PZA.`, nombres de calles, números y otros detalles de localización.
- Tipo lógico: `STRING`.
- Admite valores nulos: `SÍ`.

#### Posibles consideraciones para la integración

- conservar el texto original de la dirección;
- eliminar espacios iniciales y finales durante la normalización;
- opcionalmente normalizar abreviaturas habituales de tipos de vía;
- marcar valores excepcionalmente cortos o sospechosos para su revisión si se requiere una mayor calidad de las direcciones.

In [75]:
df["nombre"].dtype

dtype('O')

In [76]:

df["nombre"].isna().sum()

np.int64(0)

In [77]:

df["nombre"].nunique()

1660

In [14]:
with pd.option_context("display.max_colwidth", None):
    display(df["nombre"].head(15))

0                                 CENTRO CIVICO-SOCIAL "VIRGEN DEL PILAR"
1                                  CENTRO SOCIAL DE SAN MIGUEL DE SALINAS
2                            CLUB DE CONVIVENCIA DE TERCERA EDAD EL CISNE
3                                     RESIDENCIA P.M.D. NTRA.SRA.DEL PUIG
4               CENTRO SOCIAL PLA SECTOR V (EN PROCESO DE REGULARIZACION)
5                                               RESIDENCIA MIXTA MONDUBER
6                                               RESIDENCIA MIXTA MONDUBER
7     RESIDENCIA PARA PERSONAS MAYORES DEPENDIENTES JARDINES DEL PARTERRE
8                                      CENTRO SOCIAL JOSE LLORCA LLINARES
9                                           VIVIENDA TUTELADA EL RINCONET
10                                      CENTRO SOCIAL DE RAFOL DE ALMUNIA
11                   CENTRO SOCIAL LA HOYA (EN PROCESO DE REGULARIZACION)
12                        CLUB DE CONVIVENCIA DE TERCERA EDAD DEL CAMPELL
13                           CLUB DE C

In [82]:
df["nombre"].value_counts().head(30)

nombre
CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN SITUACIÓN DE DEPENDENCIA GRAN VIA                                                          7
RESIDENCIA MIXTA CENTRO GERIÁTRICO  LA MORENICA                                                                                         4
Alojamiento Alternativo Puente                                                                                                          4
RESIDENCIA PARA PERSONAS MAYORES DEPENDIENTES AYORA                                                                                     3
CENTRO RESIDENCIAL PARA PERSONAS MAYORES HOGAR SANTA TERESA JORNET                                                                      2
RESIDENCIA MIXTA DOMUSVI MONTEARSE                                                                                                      2
RESIDENCIA  MIXTA SAN LORENZO DE BRINDIS                                                                                                2
CLUB DE CONVIVENCIA MUNICIP

In [83]:
df["nombre"].duplicated(keep=False).sum()

np.int64(64)

In [84]:
name_counts = df["nombre"].value_counts()

(name_counts > 1).sum()

np.int64(27)

In [85]:
duplicate_names = df["nombre"].duplicated(keep=False)

df.loc[
    duplicate_names,
    ["id", "nombre", "municipio", "domici", "entidad"]
].sort_values("nombre")

,id,nombre,municipio,domici,entidad
1482,1483,Alojamiento Alternativo Puente,Castelló de la Plana,"C/ Hermanos Vilafañe 33,2,pta 3",HIJAS DE LA CARIDAD DE SAN VICENTE PAUL PRO. P...
1475,1476,Alojamiento Alternativo Puente,Castelló de la Plana,ALCALDE TARREGA,HIJAS DE LA CARIDAD DE SAN VICENTE PAUL PRO. P...
1457,1458,Alojamiento Alternativo Puente,Castelló de la Plana,"calle JUAN DE LA CIERVA, 2 -6 DCHA",HIJAS DE LA CARIDAD DE SAN VICENTE PAUL PRO. P...
1456,1457,Alojamiento Alternativo Puente,Castelló de la Plana,"calle JUAN DE LA CIERVA, 2- 3 DCHA",HIJAS DE LA CARIDAD DE SAN VICENTE PAUL PRO. P...
960,961,CENTRE INTEGRAL DE MAJORS,Ontinyent,AVINGUDA DE VICENTE GIRONES MORA Nº 18,AJUNTAMENT D'ONTINYENT
...,...,...,...,...,...
1694,1706,SAN JOSÉ,Castelló de la Plana,Camino del Caminás 42,"INSOEME, S.L."
1673,1683,VIVIENDA TUTELADA NOVAEDAT ALBERIC,Alberic,"CALLE ARCIPRESTE FERRANDIS, 12",NOVAEDAT BENESTAR SLU
1660,1662,VIVIENDA TUTELADA NOVAEDAT ALBERIC,Alberic,"CALLE ARCIPRESTE FERRANDIS, 12",NOVAEDAT BENESTAR SLU
1662,1664,VIVIENDA TUTELADA SUPERVISADA LLUERNA,Paiporta,C/ DR. MARAÑÓN N.º 4 Bajo,"Mistral Acció Social, Coop. V."


In [86]:
with pd.option_context(
    "display.max_rows", None,
    "display.max_colwidth", None
):
    display(
        df.loc[
            duplicate_names,
            ["id", "nombre", "municipio", "domici", "entidad"]
        ].sort_values(["nombre", "id"])
    )

,id,nombre,municipio,domici,entidad
1456,1457,Alojamiento Alternativo Puente,Castelló de la Plana,"calle JUAN DE LA CIERVA, 2- 3 DCHA",HIJAS DE LA CARIDAD DE SAN VICENTE PAUL PRO. PAMPLONA. CODAD EN OBRA SOCIAL MARILLAC
1457,1458,Alojamiento Alternativo Puente,Castelló de la Plana,"calle JUAN DE LA CIERVA, 2 -6 DCHA",HIJAS DE LA CARIDAD DE SAN VICENTE PAUL PRO. PAMPLONA. CODAD EN OBRA SOCIAL MARILLAC
1475,1476,Alojamiento Alternativo Puente,Castelló de la Plana,ALCALDE TARREGA,HIJAS DE LA CARIDAD DE SAN VICENTE PAUL PRO. PAMPLONA. CODAD EN OBRA SOCIAL MARILLAC
1482,1483,Alojamiento Alternativo Puente,Castelló de la Plana,"C/ Hermanos Vilafañe 33,2,pta 3",HIJAS DE LA CARIDAD DE SAN VICENTE PAUL PRO. PAMPLONA. CODAD EN OBRA SOCIAL MARILLAC
960,961,CENTRE INTEGRAL DE MAJORS,Ontinyent,AVINGUDA DE VICENTE GIRONES MORA Nº 18,AJUNTAMENT D'ONTINYENT
1467,1468,CENTRE INTEGRAL DE MAJORS,Llutxent,SAN ANTONI,AJUNTAMENT DE LLUTXENT
1393,1394,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN SITUACIÓN DE DEPENDENCIA GRAN VIA,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA
1394,1395,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN SITUACIÓN DE DEPENDENCIA GRAN VIA,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA
1395,1396,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN SITUACIÓN DE DEPENDENCIA GRAN VIA,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA
1396,1397,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN SITUACIÓN DE DEPENDENCIA GRAN VIA,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA


In [87]:
columns_without_id = df.columns.drop("id")

In [88]:
exact_duplicates = df.duplicated(
    subset=columns_without_id,
    keep=False
)

exact_duplicates.sum()

np.int64(0)

In [89]:
df.loc[exact_duplicates].sort_values("nombre")

,WKT,id,ni_centro,tipo_centro,nombre,max,entidad,domici,leyenda,cod_ine_mun,...,comarca,municipio,sector,clasificacion,pobtotal,email,telefono,f_resolu,x_25830,y_25830


In [90]:
visible_duplicates = df.duplicated(
    subset=["nombre", "municipio", "domici", "entidad"],
    keep=False
)

visible_duplicates.sum()

np.int64(52)

In [91]:
df.loc[
    visible_duplicates,
    ["id", "nombre", "municipio", "domici", "entidad"]
].sort_values(["nombre", "id"])

,id,nombre,municipio,domici,entidad
1393,1394,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA
1394,1395,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA
1395,1396,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA
1396,1397,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA
1397,1398,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA
1398,1399,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA
1399,1400,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,Castelló de la Plana,"AVDA TARREGA MONTEBLANCO, 43",GENERALITAT VALENCIANA
1655,1656,CENTRO DE DÍA PARA PERSONAS MAYORES DEPENDIENTES,Benaguasil,Avda. diputación esquina Camí Fondo,GENERALITAT VALENCIANA
1685,1697,CENTRO DE DÍA PARA PERSONAS MAYORES DEPENDIENTES,Benaguasil,Avda. diputación esquina Camí Fondo,GENERALITAT VALENCIANA
1059,1060,CENTRO DE TIPOLOGÍA MIXTA DE QUART DE POBLET,Quart de Poblet,"AVENIDA DE MADRID, 8",AJUNTAMENT DE QUART DE POBLET


In [93]:
key_cols = ["nombre", "municipio", "domici", "entidad"]

comparison_cols = [
    col for col in df.columns
    if col not in key_cols + ["id"]
]

group_variation = (
    df.loc[visible_duplicates]
    .groupby(key_cols, dropna=False)[comparison_cols]
    .nunique(dropna=False)
)

In [94]:
difference_summary = (
    (group_variation > 1)
    .sum()
    .sort_values(ascending=False)
)

difference_summary[difference_summary > 0]

clasificacion    22
max              19
sector           12
ni_centro        10
f_resolu         10
pobtotal         10
leyenda          10
tipo_centro       9
dtype: int64

### `nombre` — conclusion

- The field is complete: all 1697 records contain a value.
- There are 1660 distinct names.
- 27 names occur more than once, affecting 64 records.
- Repeated names do not necessarily represent duplicate records. In several cases, the same centre appears in different records because other attributes, such as classification, sector or centre type, differ.
- No completely duplicated rows were found when the `id` field was excluded.
- The field is suitable for the global attribute `Entidad.nombre`.
- It should not be used as a unique identifier because the same name may appear in multiple records.

### `nombre` — conclusión

- El campo está completo: los 1697 registros contienen un valor.
- Hay 1660 nombres distintos.
- 27 nombres aparecen más de una vez, afectando a 64 registros.
- Los nombres repetidos no representan necesariamente registros duplicados. En varios casos, el mismo centro aparece en diferentes registros porque cambian otros atributos, como la clasificación, el sector o el tipo de centro.
- No se encontraron filas completamente duplicadas al excluir el campo `id`.
- El campo es adecuado para el atributo global `Entidad.nombre`.
- No debe utilizarse como identificador único, ya que un mismo nombre puede aparecer en varios registros.

In [4]:
df["max"].dtype

dtype('float64')

In [5]:
df["max"].isna().sum()

np.int64(32)

In [6]:
df["max"].notna().sum()

np.int64(1665)

In [7]:
df["max"].head(15)

0      0.0
1      0.0
2      0.0
3     31.0
4      0.0
5     80.0
6     88.0
7     24.0
8      0.0
9      7.0
10     0.0
11     0.0
12     0.0
13     0.0
14     0.0
Name: max, dtype: float64

In [8]:
df["max"].nunique()

147

In [9]:
df["max"].value_counts()

max
0.0      395
7.0      124
40.0      95
70.0      70
6.0       65
        ... 
123.0      1
159.0      1
108.0      1
103.0      1
13.0       1
Name: count, Length: 147, dtype: int64

In [10]:
df["max"].describe()

count    1665.000000
mean       35.167568
std        41.902705
min         0.000000
25%         4.000000
50%        20.000000
75%        52.000000
max       356.000000
Name: max, dtype: float64

In [11]:
(df["max"].dropna() % 1 != 0).sum()


np.int64(0)

In [12]:
df["max"].isna().sum()

np.int64(32)

In [13]:
df["max"].notna().sum()

np.int64(1665)

In [15]:
df.loc[df["max"].isna(), "clasificacion"].value_counts(dropna=False)

clasificacion
SIN RESOLUCIÓN    32
Name: count, dtype: int64

In [16]:
df.loc[df["max"].isna(), "clasificacion"].unique()

array(['SIN RESOLUCIÓN'], dtype=object)

### `max` — conclusion

- The field is stored as `float64`.
- It contains 1665 non-null values and 32 missing values.
- All non-null values are mathematically integers, so a nullable integer type would better represent the data.
- The field contains 147 distinct values.
- Values range from 0 to 356, with a median of 20.
- 395 records contain the value `0`.
- The 32 missing values occur in exactly the same records where `ni_centro` and `pobtotal` are also missing.
- All records with a missing `max` value have `clasificacion = "SIN RESOLUCIÓN"`.
- The values suggest that the field may represent a maximum capacity or number of places, but the meaning cannot be confirmed from the column name and data alone.
- Therefore, the field should not yet be mapped to the global schema unless its semantic meaning is confirmed.

### `max` — conclusión

- El campo está almacenado como `float64`.
- Contiene 1665 valores no nulos y 32 valores ausentes.
- Todos los valores no nulos son matemáticamente enteros, por lo que un tipo entero anulable representaría mejor los datos.
- El campo contiene 147 valores distintos.
- Los valores van de 0 a 356, con una mediana de 20.
- 395 registros contienen el valor `0`.
- Los 32 valores ausentes aparecen exactamente en los mismos registros en los que también faltan `ni_centro` y `pobtotal`.
- Todos los registros con `max` ausente tienen `clasificacion = "SIN RESOLUCIÓN"`.
- Los valores sugieren que el campo podría representar una capacidad máxima o un número de plazas, pero su significado no puede confirmarse únicamente a partir del nombre de la columna y de los datos.
- Por tanto, el campo todavía no debe asociarse con la estructura global hasta confirmar su significado semántico.

In [17]:
df["entidad"].dtype

dtype('O')

In [18]:
df["entidad"].notna().sum()

np.int64(1697)

In [25]:
with pd.option_context(
    "display.max_colwidth", None
):display(df["entidad"].head(15))

0                                                          AYUNTAMIENTO DE LOS MONTESINOS
1                                                   AYUNTAMIENTO DE SAN MIGUEL DE SALINAS
2                                                                    AJUNTAMENT D'ALACANT
3                                                                      AMPARO CAMPS, S.L.
4                                                                        AJUNTAMENT D'ELX
5                                             EL CASTILLO ATENCIÓN A LA DEPENDENCIA, S.A.
6                                             EL CASTILLO ATENCIÓN A LA DEPENDENCIA, S.A.
7                                                CENTRO GERIATR.JARDINES DEL PARTERRE C.B
8                                                                  AJUNTAMENT DE BENIDORM
9                                                     ASOCIACION GRANJA-HOGAR EL RINCONET
10                                                         AJUNTAMENT DE RAFOL DE ALMUNIA
11        

In [26]:
df["entidad"].nunique()

725

In [27]:
df["entidad"].value_counts()

entidad
GENERALITAT VALENCIANA                                                      141
INSTITUTO VALENCIANO DE SERVICIOS SOCIALES                                   40
NOVAEDAT BENESTAR SLU                                                        26
CENTROS RESIDENCIALES SAVIA, SL                                              26
FUNDACIÓ ESPURNA DE LA C.V.                                                  25
                                                                           ... 
FUVANE, FUNDACION DE LA COMUNIDAD VALENCIANA PARA LA NEUROREHABILITACION      1
PELLERANO, S.L.                                                               1
AJUNTAMENT D'ALBALAT DE LA RIBERA                                             1
AJUNTAMENT DE XERT                                                            1
AYUNTAMIENTO DE GAIBIEL                                                       1
Name: count, Length: 725, dtype: int64

In [28]:
df.groupby("entidad")["nombre"].nunique().sort_values(ascending=False).head(20)

entidad
GENERALITAT VALENCIANA                                                                        134
INSTITUTO VALENCIANO DE SERVICIOS SOCIALES                                                     40
CENTROS RESIDENCIALES SAVIA, SL                                                                26
NOVAEDAT BENESTAR SLU                                                                          25
FUNDACIÓ ESPURNA DE LA C.V.                                                                    25
QUAVITAE SERVICIOS ASISTENCIALES, S.A.                                                         23
LA SALETA CARE SL                                                                              23
AJUNTAMENT DE VALENCIA                                                                         16
AJUNTAMENT DE CASTELLO DE LA PLANA                                                             15
SANT JOAN DE DEU SERVEIS SOCIALS VALENCIA HOSPITALARIOS DE SAN JUAN DE DIOS-PROV DE ARAGON     15
HERMANITAS D

In [29]:
df.groupby("nombre")["entidad"].nunique().sort_values(ascending=False).head(20)

nombre
CENTRE INTEGRAL DE MAJORS                                     2
CLUB DE CONVIVENCIA MUNICIPAL                                 2
vivienda tutelada ACOFEM-13                                   1
ABADIA TURIA                                                  1
ACCEM                                                         1
ACCEM - V.T. para personas sin hogar o exc. social Linares    1
ACCEM - VIVIENDA AVENIDA PÉREZ GALDÓS                         1
ACCEM - Vivienda de Acogida Temporal Buenos Casalduch         1
ACCEM- Vivienda de Acogida Temporal Gadea                     1
ACCEM- Vivienda de Acogida Temporal Gaspar Aguilar            1
ACCEM- Vivienda de Acogida Temporal Herrero                   1
ACCEM- Vivienda de Acogida Temporal Maluquer                  1
ACCEM- Vivienda de Acogida Temporal Manuel Gago               1
Vivienda Regina Mas SJDV                                      1
Vivienda Supervisada Atalaya                                  1
Vivienda Tutelada  CIUDAD JARDIN 

In [30]:
nombre_multiple_entidades = (
    df.groupby("nombre")["entidad"]
      .nunique()
      .loc[lambda x: x > 1]
      .sort_values(ascending=False)
)

nombre_multiple_entidades

nombre
CENTRE INTEGRAL DE MAJORS        2
CLUB DE CONVIVENCIA MUNICIPAL    2
Name: entidad, dtype: int64

In [31]:
problem_names = (
    df.groupby("nombre")["entidad"]
      .nunique()
      .loc[lambda x: x > 1]
      .index
)

df.loc[
    df["nombre"].isin(problem_names),
    [
        "nombre",
        "entidad",
        "ni_centro",
        "tipo_centro",
        "domici",
        "municipio",
        "sector",
        "clasificacion"
    ]
].sort_values(["nombre", "entidad"])

,nombre,entidad,ni_centro,tipo_centro,domici,municipio,sector,clasificacion
960,CENTRE INTEGRAL DE MAJORS,AJUNTAMENT D'ONTINYENT,3392.0,CENTROS INTEGRALES DE MAYORES (CIMS) DE TITULA...,AVINGUDA DE VICENTE GIRONES MORA Nº 18,Ontinyent,PERSONAS MAYORES,PERSONAS MAYORES
1467,CENTRE INTEGRAL DE MAJORS,AJUNTAMENT DE LLUTXENT,11079.0,CENTROS INTEGRALES DE MAYORES (CIMS) DE TITULA...,SAN ANTONI,Llutxent,PERSONAS MAYORES,PERSONAS MAYORES
76,CLUB DE CONVIVENCIA MUNICIPAL,AJUNTAMENT DE FONTANARS DELS ALFORINS,413.0,CLUBS DE CONVIVENCIA PARA MAYORES,"CONDE SALVATIERRA, 29",Fontanars dels Alforins,PERSONAS MAYORES,PERSONAS MAYORES
370,CLUB DE CONVIVENCIA MUNICIPAL,AYUNTAMIENTO DE MONTAN,1386.0,HOGARES Y CLUBS PARA PERSONAS MAYORES,AVDA. ANTONIO FORNAS Nº 10,Montán,PERSONAS MAYORES,PERSONAS MAYORES


In [32]:
df[df["ni_centro"] == 10233][
    ["id", "ni_centro", "nombre", "entidad", "sector", "tipo_centro"]
]

,id,ni_centro,nombre,entidad,sector,tipo_centro
1393,1394,10233.0,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,GENERALITAT VALENCIANA,PERSONAS CON DISCAPACIDAD,CENTRO DE DÍA PARA PERSONAS CON DISCAPACIDAD O...
1394,1395,10233.0,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,GENERALITAT VALENCIANA,PERSONAS CON ENFERMEDAD MENTAL,CENTRO DE DÍA PARA PERSONAS CON ENFERMEDAD MEN...
1395,1396,10233.0,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,GENERALITAT VALENCIANA,PERSONAS CON ENFERMEDAD MENTAL,CENTRO DE REHABILITACIÓN E INTEGRACIÓN SOCIAL ...
1396,1397,10233.0,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,GENERALITAT VALENCIANA,PERSONAS CON ENFERMEDAD MENTAL,CENTRO ESPECÍFICO PARA ENFERMOS MENTALES CRÓNI...
1397,1398,10233.0,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,GENERALITAT VALENCIANA,PERSONAS MAYORES,CENTROS DE DÍA DE PERSONAS MAYORES DEPENDIENTES
1398,1399,10233.0,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,GENERALITAT VALENCIANA,PERSONAS CON DISCAPACIDAD,RESIDENCIA DE PERSONAS CON DISCAPACIDAD O DIVE...
1399,1400,10233.0,CENTRO DE ATENCIÓN ESPECIAL PARA PERSONAS EN S...,GENERALITAT VALENCIANA,PERSONAS MAYORES,RESIDENCIAS PARA PERSONAS MAYORES DEPENDIENTES


### `entidad` — conclusion

- The field represents the organisation or institution associated with the centre, rather than the centre itself.
- A single `entidad` can be associated with many different centres (`nombre`).
- For example, `GENERALITAT VALENCIANA` appears in 141 records and is associated with 134 distinct centre names.
- In the opposite direction, almost every centre name is associated with only one `entidad`.
- Only two centre names are associated with more than one entity: `CENTRE INTEGRAL DE MAJORS` and `CLUB DE CONVIVENCIA MUNICIPAL`.
- Inspection of these cases shows that they correspond to different centres with different `ni_centro`, municipalities and addresses. Therefore, the repeated names do not indicate that one centre belongs to multiple entities.
- The field is therefore useful for identifying the organisation responsible for or associated with a centre.
- It should not be confused with `nombre`, which identifies the centre itself.
- Its mapping to the global schema depends on whether the global model includes the organisation responsible for the centre as a separate concept.

### `entidad` — conclusión

- El campo representa la organización o institución asociada al centro, y no el propio centro.
- Una misma `entidad` puede estar asociada a muchos centros diferentes (`nombre`).
- Por ejemplo, `GENERALITAT VALENCIANA` aparece en 141 registros y está asociada a 134 nombres de centro distintos.
- En la dirección inversa, casi todos los nombres de centro están asociados a una sola `entidad`.
- Solo dos nombres aparecen asociados a más de una entidad: `CENTRE INTEGRAL DE MAJORS` y `CLUB DE CONVIVENCIA MUNICIPAL`.
- La inspección de estos casos muestra que corresponden a centros diferentes, con distintos `ni_centro`, municipios y direcciones. Por tanto, los nombres repetidos no indican que un mismo centro pertenezca a varias entidades.
- El campo es útil para identificar la organización responsable o asociada al centro.
- No debe confundirse con `nombre`, que identifica al propio centro.
- Su correspondencia con la estructura global depende de si el modelo global contempla por separado la organización responsable del centro.

In [33]:
df["leyenda"].dtype

dtype('O')

In [34]:
df["leyenda"].isna().sum()

np.int64(0)

In [35]:
df["leyenda"].notna().sum()

np.int64(1697)

In [36]:
df["leyenda"].nunique()

9

In [37]:
df.groupby("entidad")["leyenda"].nunique().sort_values(ascending=False).head(20)

entidad
AJUNTAMENT DE VILA-REAL                                                                              2
AJUNTAMENT DE VALENCIA                                                                               2
ASOCIACIÓN PRO-PERSONAS CON DISCAPACIDAD DE ASPE                                                     2
AJUNTAMENT DE CARCAIXENT                                                                             2
ASOC. PARA LA ATENCION DE LAS PERSONAS CON DISCAPACIDAD INTELECTUAL DE VILLENA Y COMARCA - APADIS    2
AJUNTAMENT DE MONCOFAR                                                                               2
AJUNTAMENT DE LA POBLA DE VALLBONA                                                                   2
AJUNTAMENT DE BENICASSIM                                                                             2
AJUNTAMENT DE SUECA                                                                                  2
AJUNTAMENT DE SANTA POLA                                         

In [38]:
df.groupby("entidad")["leyenda"].nunique().value_counts()

leyenda
1    704
2     21
Name: count, dtype: int64

In [39]:
leyenda_multiple = (
    df.groupby("entidad")["leyenda"]
      .nunique()
)

leyenda_multiple.loc[leyenda_multiple > 1]

entidad
AJUNTAMENT DE BENICASSIM                                                                             2
AJUNTAMENT DE CARCAIXENT                                                                             2
AJUNTAMENT DE LA POBLA DE VALLBONA                                                                   2
AJUNTAMENT DE MONCOFAR                                                                               2
AJUNTAMENT DE SANTA POLA                                                                             2
AJUNTAMENT DE SUECA                                                                                  2
AJUNTAMENT DE VALENCIA                                                                               2
AJUNTAMENT DE VILA-REAL                                                                              2
ASOC. PARA LA ATENCION DE LAS PERSONAS CON DISCAPACIDAD INTELECTUAL DE VILLENA Y COMARCA - APADIS    2
ASOCIACIÓN PRO-PERSONAS CON DISCAPACIDAD DE ASPE                 

In [40]:
leyenda_count = df.groupby("entidad")["leyenda"].nunique()

entidades_multi_leyenda = leyenda_count[leyenda_count > 1].index

df.loc[
    df["entidad"].isin(entidades_multi_leyenda),
    ["entidad", "nombre", "ni_centro", "leyenda"]
].sort_values(["entidad", "leyenda"])

,entidad,nombre,ni_centro,leyenda
67,AJUNTAMENT DE BENICASSIM,CLUB DE CONVIVENCIA LA GARROFERA,324.0,Entidad Local
1653,AJUNTAMENT DE BENICASSIM,Centro de Día Benicàssim,11332.0,Entidad Local
1654,AJUNTAMENT DE BENICASSIM,CIM Benicàssim,11333.0,Entidad Local
1686,AJUNTAMENT DE BENICASSIM,Centro de Día Benicàssim,NaN,SIN INFORMAR
1687,AJUNTAMENT DE BENICASSIM,CIM Benicàssim,NaN,SIN INFORMAR
...,...,...,...,...
1585,NOVAEDAT BENESTAR SLU,VIVIENDA TUTELADA NOVAEDAT BENIARBEIG,11245.0,Entidad Privada
1660,NOVAEDAT BENESTAR SLU,VIVIENDA TUTELADA NOVAEDAT ALBERIC,11343.0,Entidad Privada
1673,NOVAEDAT BENESTAR SLU,VIVIENDA TUTELADA NOVAEDAT ALBERIC,NaN,OTRAS
389,RESIDENCIAL CASTELLON SERVICIOS INTEGRALES TER...,RESIDENCIAL PARA PERSONAS MAYORES DEPENDIENTES...,1429.0,Entidad Privada


In [41]:
df.loc[df["entidad"].isin(entidades_multi_leyenda)] \
  .groupby("entidad")["leyenda"] \
  .unique()

entidad
AJUNTAMENT DE BENICASSIM                                                                                       [Entidad Local, SIN INFORMAR]
AJUNTAMENT DE CARCAIXENT                                                                                              [Entidad Local, OTRAS]
AJUNTAMENT DE LA POBLA DE VALLBONA                                                                                    [Entidad Local, OTRAS]
AJUNTAMENT DE MONCOFAR                                                                                                [Entidad Local, OTRAS]
AJUNTAMENT DE SANTA POLA                                                                                              [Entidad Local, OTRAS]
AJUNTAMENT DE SUECA                                                                                                   [Entidad Local, OTRAS]
AJUNTAMENT DE VALENCIA                                                                                                [Entidad Local, OTRAS]
AJUNT

### `leyenda` — conclusion

- The field is categorical and contains 9 distinct values.
- Most organisations have a single `leyenda` value: 704 entities have one category.
- 21 entities are associated with two different `leyenda` values.
- The combinations show that the field mixes categories at different levels of specificity, for example:
  - `Entidad Privada` / `EMPRESA Y SOCIEDADES`
  - `Entidad Privada` / `FUNDACION`
  - `Entidad Privada` / `COOPERATIVA`
  - `Entidad Local` / `ENTIDAD PÚBLICA`
  - `Entidad Propia` / `ENTIDAD PÚBLICA`
- Therefore, `leyenda` should not be interpreted as a strict or unique type of organisation.
- It appears to represent a source-specific administrative or legal classification of the entity.
- The field has no direct equivalent in the current global schema, so it should not be mapped unless a specific attribute for ownership, legal nature or entity type is added.

### `leyenda` — conclusión

- El campo es categórico y contiene 9 valores distintos.
- La mayoría de las organizaciones tienen un único valor de `leyenda`: 704 entidades tienen una sola categoría.
- 21 entidades están asociadas a dos valores diferentes de `leyenda`.
- Las combinaciones muestran que el campo mezcla categorías con distintos niveles de especificidad, por ejemplo:
  - `Entidad Privada` / `EMPRESA Y SOCIEDADES`
  - `Entidad Privada` / `FUNDACION`
  - `Entidad Privada` / `COOPERATIVA`
  - `Entidad Local` / `ENTIDAD PÚBLICA`
  - `Entidad Propia` / `ENTIDAD PÚBLICA`
- Por tanto, `leyenda` no debe interpretarse como un tipo de organización único y estricto.
- Parece representar una clasificación administrativa o jurídica específica de la fuente.
- El campo no tiene una correspondencia directa con la estructura global actual, por lo que no debería mapearse salvo que se añada un atributo específico para titularidad, naturaleza jurídica o tipo de entidad.

### `provincia` — conclusion

- The field contains three distinct provinces: `Alacant/Alicante`, `Castelló/Castellón` and `València/Valencia`.
- All 1697 records have a province value.
- The province is fully consistent with the first two digits of `cod_ine_mun`:
  - `03` → `Alacant/Alicante`
  - `12` → `Castelló/Castellón`
  - `46` → `València/Valencia`
- The relationship with `comarca` is also consistent: all 33 comarcas are associated with exactly one province.
- No geographical inconsistencies were detected between `provincia`, `comarca` and `cod_ine_mun`.
- The field can be mapped directly to `Provincia.nombre` in the global schema.
- `Provincia.codigo` can be derived from the first two digits of `cod_ine_mun`.
- Province names are stored in bilingual Valencian/Spanish form, which should be taken into account during normalization.

### `provincia` — conclusión

- El campo contiene tres provincias distintas: `Alacant/Alicante`, `Castelló/Castellón` y `València/Valencia`.
- Los 1697 registros contienen un valor de provincia.
- La provincia es totalmente coherente con los dos primeros dígitos de `cod_ine_mun`:
  - `03` → `Alacant/Alicante`
  - `12` → `Castelló/Castellón`
  - `46` → `València/Valencia`
- La relación con `comarca` también es coherente: las 33 comarcas están asociadas a una única provincia.
- No se detectaron inconsistencias geográficas entre `provincia`, `comarca` y `cod_ine_mun`.
- El campo puede asociarse directamente con `Provincia.nombre` en la estructura global.
- `Provincia.codigo` puede obtenerse a partir de los dos primeros dígitos de `cod_ine_mun`.
- Los nombres de provincia están almacenados en formato bilingüe valenciano/castellano, lo que debe tenerse en cuenta durante la normalización.

In [44]:
df.groupby("ni_centro")["sector"].nunique().value_counts()

sector
1    1633
2      10
3       2
Name: count, dtype: int64

In [50]:
sector_counts = df.groupby("ni_centro")["sector"].nunique()
multiple_sectors = sector_counts[sector_counts >1]
multiple_sectors

ni_centro
50.0       2
997.0      2
998.0      2
1412.0     3
1849.0     2
2804.0     2
3513.0     2
3592.0     2
4084.0     2
4085.0     2
10233.0    3
11145.0    2
Name: sector, dtype: int64

In [53]:
with pd.option_context(
    "display.max_colwidth", None
):
    display(df.loc[
        df["ni_centro"].isin(multiple_sectors.index),
        ["ni_centro", "nombre", "sector", "clasificacion", "tipo_centro"]
    ])

,ni_centro,nombre,sector,clasificacion,tipo_centro
5,50.0,RESIDENCIA MIXTA MONDUBER,PERSONAS CON ENFERMEDAD MENTAL,PERSONAS CON ENFERMEDAD MENTAL,CENTRO ESPECÍFICO PARA ENFERMOS MENTALES CRÓNICOS (CEEM)
6,50.0,RESIDENCIA MIXTA MONDUBER,PERSONAS MAYORES,PERSONAS MAYORES,RESIDENCIAS PARA PERSONAS MAYORES DEPENDIENTES
201,997.0,RESIDENCIA MIXTA DOMUSVI MONTEARSE,PERSONAS CON ENFERMEDAD MENTAL,PERSONAS CON ENFERMEDAD MENTAL,CENTRO ESPECÍFICO PARA ENFERMOS MENTALES CRÓNICOS (CEEM)
202,997.0,RESIDENCIA MIXTA DOMUSVI MONTEARSE,PERSONAS MAYORES,PERSONAS MAYORES,RESIDENCIAS PARA PERSONAS MAYORES DEPENDIENTES
203,998.0,RESIDENCIA MIXTA SAN LORENZO DE BRINDIS,PERSONAS CON ENFERMEDAD MENTAL,PERSONAS CON ENFERMEDAD MENTAL,CENTRO ESPECÍFICO PARA ENFERMOS MENTALES CRÓNICOS (CEEM)
204,998.0,RESIDENCIA MIXTA SAN LORENZO DE BRINDIS,PERSONAS MAYORES,PERSONAS MAYORES,RESIDENCIAS PARA PERSONAS MAYORES DEPENDIENTES
377,1412.0,RESIDENCIA MIXTA CENTRO GERIÁTRICO LA MORENICA,PERSONAS CON ENFERMEDAD MENTAL,PERSONAS CON ENFERMEDAD MENTAL,CENTRO ESPECÍFICO PARA ENFERMOS MENTALES CRÓNICOS (CEEM)
378,1412.0,RESIDENCIA MIXTA CENTRO GERIÁTRICO LA MORENICA,PERSONAS CON DISCAPACIDAD,PERSONAS CON DISCAPACIDAD,RESIDENCIAS PARA PERSONAS CON DIVERSIDAD FUNCIONAL FÍSICA
379,1412.0,RESIDENCIA MIXTA CENTRO GERIÁTRICO LA MORENICA,PERSONAS CON DISCAPACIDAD,PERSONAS CON DISCAPACIDAD,RESIDENCIAS PARA PERSONAS CON DIVERSIDAD FUNCIONAL PSÍQUICA
380,1412.0,RESIDENCIA MIXTA CENTRO GERIÁTRICO LA MORENICA,PERSONAS MAYORES,PERSONAS MAYORES,RESIDENCIAS PARA PERSONAS MAYORES DEPENDIENTES


### `sector` — conclusion

- The field represents the target population or social-service area of the centre.
- It is a strong candidate for mapping to `Entidad.ambito` in the global schema.
- Most centres are associated with a single sector.
- However, 12 centres are associated with more than one sector: 10 centres have two sectors and 2 centres have three.
- Inspection of these cases shows that this is meaningful information rather than duplicated data: the same centre can provide services to different target groups.
- Therefore, the relationship between `ni_centro` and `sector` is not strictly one-to-one.
- This must be taken into account when mapping `sector` to the global `ambito` attribute, since one centre may require several `ambito` values.

### `sector` — conclusión

- El campo representa la población destinataria o el ámbito de actuación social del centro.
- Es un candidato claro para corresponder con `Entidad.ambito` en la estructura global.
- La mayoría de los centros están asociados a un único sector.
- Sin embargo, 12 centros están asociados a más de un sector: 10 centros tienen dos sectores y 2 centros tienen tres.
- La inspección de estos casos muestra que no se trata de datos duplicados, sino de información significativa: un mismo centro puede prestar servicios a diferentes grupos de población.
- Por tanto, la relación entre `ni_centro` y `sector` no es estrictamente uno-a-uno.
- Esto debe tenerse en cuenta al asociar `sector` con el atributo global `ambito`, ya que un mismo centro puede necesitar varios valores de `ambito`.

In [57]:
(df["clasificacion"] == df["sector"]).value_counts()

True     1665
False      32
Name: count, dtype: int64

In [58]:
df.loc[
    df["clasificacion"] != df["sector"],
    ["id", "sector", "clasificacion"]
]

,id,sector,clasificacion
1665,1667,PERSONAS CON DISCAPACIDAD,SIN RESOLUCIÓN
1666,1668,PERSONAS CON DISCAPACIDAD,SIN RESOLUCIÓN
1667,1672,MUJERES,SIN RESOLUCIÓN
1668,1673,MUJERES,SIN RESOLUCIÓN
1669,1676,PERSONAS CON DISCAPACIDAD,SIN RESOLUCIÓN
1670,1680,PERSONAS CON DISCAPACIDAD,SIN RESOLUCIÓN
1671,1681,PERSONAS CON ENFERMEDAD MENTAL,SIN RESOLUCIÓN
1672,1682,PERSONAS CON ENFERMEDAD MENTAL,SIN RESOLUCIÓN
1673,1683,PERSONAS CON ENFERMEDAD MENTAL,SIN RESOLUCIÓN
1674,1684,PERSONAS CON ENFERMEDAD MENTAL,SIN RESOLUCIÓN


In [59]:
pd.crosstab(df["sector"], df["clasificacion"])

clasificacion,PERSONAS CON DISCAPACIDAD,PERSONAS CON ENFERMEDAD MENTAL,PERSONAS MAYORES,PERSONAS MIGRANTES,POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE DESFAVORECIDOS,SIN RESOLUCIÓN
sector,,,,,,
INFANCIA,0,0,0,0,0,5
MUJERES,0,0,0,0,0,2
PERSONAS CON DISCAPACIDAD,413,0,0,0,0,6
PERSONAS CON ENFERMEDAD MENTAL,0,239,0,0,0,5
PERSONAS MAYORES,0,0,736,0,0,7
PERSONAS MIGRANTES,0,0,0,41,0,0
POBLACIÓN EN GENERAL Y COLECTIVOS SOCIALMENTE DESFAVORECIDOS,0,0,0,0,236,7


### `clasificacion` — conclusion

- The field is almost entirely redundant with `sector`.
- In 1665 of the 1697 records, `clasificacion` and `sector` contain exactly the same value.
- The remaining 32 records have `clasificacion = "SIN RESOLUCIÓN"`, while `sector` still contains a valid target-group category.
- Therefore, `clasificacion` does not provide an additional thematic classification for most records.
- Its main additional information is the special value `SIN RESOLUCIÓN`, which appears to indicate that the classification has not been formally resolved.
- For mapping to the global `Entidad.ambito` attribute, `sector` is more appropriate because it remains informative for all records.
- `clasificacion` should not be mapped separately to `ambito`.

### `clasificacion` — conclusión

- El campo es casi completamente redundante con `sector`.
- En 1665 de los 1697 registros, `clasificacion` y `sector` contienen exactamente el mismo valor.
- Los 32 registros restantes tienen `clasificacion = "SIN RESOLUCIÓN"`, mientras que `sector` sigue conteniendo una categoría válida de población destinataria.
- Por tanto, `clasificacion` no aporta una clasificación temática adicional en la mayoría de los registros.
- Su principal información adicional es el valor especial `SIN RESOLUCIÓN`, que parece indicar que la clasificación no ha sido formalmente resuelta.
- Para la correspondencia con el atributo global `Entidad.ambito`, `sector` es más adecuado porque conserva información en todos los registros.
- `clasificacion` no debería asociarse por separado con `ambito`.

In [61]:
df["pobtotal"].dtype


dtype('float64')

In [62]:
df["pobtotal"].isna().value_counts()

pobtotal
False    1665
True       32
Name: count, dtype: int64

In [63]:
(df["pobtotal"].dropna() % 1 == 0).all()

np.True_

In [64]:
df["pobtotal"].nunique()

331

In [65]:
df["pobtotal"].describe()

count      1665.000000
mean     158741.714114
std      262174.336004
min          46.000000
25%       10537.000000
50%       31246.000000
75%      170990.000000
max      790201.000000
Name: pobtotal, dtype: float64

In [67]:
df.groupby("municipio")["pobtotal"].nunique().value_counts()

pobtotal
1    333
Name: count, dtype: int64

In [68]:
df.groupby("cod_ine_mun")["pobtotal"].nunique().value_counts()

pobtotal
1    333
Name: count, dtype: int64

### `pobtotal` — conclusion

- The field is stored as `float64`, but all non-null values are mathematically integers.
- It contains 1665 non-null values and 32 missing values.
- There are 331 distinct values.
- For each `municipio`, only one unique `pobtotal` value is present.
- The same is true for each `cod_ine_mun`.
- This strongly indicates that `pobtotal` represents the total population of the municipality rather than a property of the centre.
- The field therefore belongs conceptually to the municipality/locality level.
- It has no direct equivalent in the current global schema, where `Localidad` only contains `código` and `nombre`.

### `pobtotal` — conclusión

- El campo está almacenado como `float64`, aunque todos los valores no nulos son matemáticamente enteros.
- Contiene 1665 valores no nulos y 32 valores ausentes.
- Hay 331 valores distintos.
- Para cada `municipio` existe un único valor de `pobtotal`.
- Lo mismo ocurre para cada `cod_ine_mun`.
- Esto indica con bastante claridad que `pobtotal` representa la población total del municipio y no una característica del centro.
- Por tanto, conceptualmente pertenece al nivel de municipio/localidad.
- No tiene una correspondencia directa en la estructura global actual, donde `Localidad` solo contiene `código` y `nombre`.

In [69]:
df.groupby("ni_centro")["tipo_centro"].nunique().value_counts()

tipo_centro
1    1636
2       6
4       1
3       1
7       1
Name: count, dtype: int64

In [83]:
tipo_counts = df.groupby("ni_centro")["tipo_centro"].nunique()
multiple_tipos = tipo_counts[tipo_counts > 1]
with pd.option_context(
    "display.max_colwidth", None
):
    display(df.loc[
        df["ni_centro"].isin(multiple_tipos.index),
        ["ni_centro", "sector", "nombre", "tipo_centro"]
    ])

,ni_centro,sector,nombre,tipo_centro
5,50.0,PERSONAS CON ENFERMEDAD MENTAL,RESIDENCIA MIXTA MONDUBER,CENTRO ESPECÍFICO PARA ENFERMOS MENTALES CRÓNICOS (CEEM)
6,50.0,PERSONAS MAYORES,RESIDENCIA MIXTA MONDUBER,RESIDENCIAS PARA PERSONAS MAYORES DEPENDIENTES
201,997.0,PERSONAS CON ENFERMEDAD MENTAL,RESIDENCIA MIXTA DOMUSVI MONTEARSE,CENTRO ESPECÍFICO PARA ENFERMOS MENTALES CRÓNICOS (CEEM)
202,997.0,PERSONAS MAYORES,RESIDENCIA MIXTA DOMUSVI MONTEARSE,RESIDENCIAS PARA PERSONAS MAYORES DEPENDIENTES
203,998.0,PERSONAS CON ENFERMEDAD MENTAL,RESIDENCIA MIXTA SAN LORENZO DE BRINDIS,CENTRO ESPECÍFICO PARA ENFERMOS MENTALES CRÓNICOS (CEEM)
204,998.0,PERSONAS MAYORES,RESIDENCIA MIXTA SAN LORENZO DE BRINDIS,RESIDENCIAS PARA PERSONAS MAYORES DEPENDIENTES
377,1412.0,PERSONAS CON ENFERMEDAD MENTAL,RESIDENCIA MIXTA CENTRO GERIÁTRICO LA MORENICA,CENTRO ESPECÍFICO PARA ENFERMOS MENTALES CRÓNICOS (CEEM)
378,1412.0,PERSONAS CON DISCAPACIDAD,RESIDENCIA MIXTA CENTRO GERIÁTRICO LA MORENICA,RESIDENCIAS PARA PERSONAS CON DIVERSIDAD FUNCIONAL FÍSICA
379,1412.0,PERSONAS CON DISCAPACIDAD,RESIDENCIA MIXTA CENTRO GERIÁTRICO LA MORENICA,RESIDENCIAS PARA PERSONAS CON DIVERSIDAD FUNCIONAL PSÍQUICA
380,1412.0,PERSONAS MAYORES,RESIDENCIA MIXTA CENTRO GERIÁTRICO LA MORENICA,RESIDENCIAS PARA PERSONAS MAYORES DEPENDIENTES


### `tipo_centro` — conclusion

- The field describes the specific type of service or resource provided by a centre.
- Most centres (1636) are associated with only one `tipo_centro`.
- However, 9 centres have more than one type:
  - 6 centres have 2 types,
  - 1 centre has 3 types,
  - 1 centre has 4 types,
  - 1 centre has 7 types.
- These cases show that one physical centre (`ni_centro`) can provide several different types of services.
- `tipo_centro` is more specific than `sector`: a centre may have several `tipo_centro` values within the same or different sectors.
- Therefore, `tipo_centro` should be interpreted as a service/resource subtype rather than as an identifier or general category of the centre.
- The current global schema has no direct attribute corresponding to this field.

### `tipo_centro` — conclusión

- El campo describe el tipo específico de servicio o recurso que presta un centro.
- La mayoría de los centros (1636) están asociados a un único `tipo_centro`.
- Sin embargo, 9 centros tienen más de un tipo:
  - 6 centros tienen 2 tipos,
  - 1 centro tiene 3 tipos,
  - 1 centro tiene 4 tipos,
  - 1 centro tiene 7 tipos.
- Estos casos muestran que un mismo centro físico (`ni_centro`) puede prestar varios tipos diferentes de servicios.
- `tipo_centro` es más específico que `sector`: un centro puede tener varios valores de `tipo_centro` dentro del mismo sector o de sectores diferentes.
- Por tanto, `tipo_centro` debe interpretarse como un subtipo de servicio o recurso, y no como identificador o categoría general del centro.
- La estructura global actual no contiene un atributo que corresponda directamente a este campo.

In [87]:
df["ni_centro"].isna().value_counts()

ni_centro
False    1665
True       32
Name: count, dtype: int64

In [88]:
(df["ni_centro"].dropna() % 1 == 0 ).all()

np.True_

In [89]:
df.groupby("ni_centro")["nombre"].nunique().value_counts()

nombre
1    1645
Name: count, dtype: int64

In [93]:
comparison = (
    df["ni_centro"].isna()
    ==
    (df["clasificacion"] == "SIN RESOLUCIÓN")
)

comparison.value_counts()

True    1697
Name: count, dtype: int64

In [94]:
missing_ni = df["ni_centro"].isna()
sin_resolucion = df["clasificacion"] == "SIN RESOLUCIÓN"

pd.crosstab(
    missing_ni,
    sin_resolucion,
    rownames=["ni_centro is missing"],
    colnames=["clasificacion is SIN RESOLUCIÓN"]
)

clasificacion is SIN RESOLUCIÓN,False,True
ni_centro is missing,,
False,1665,0
True,0,32


In [95]:
missing_ni = df["ni_centro"].isna()
isna = df["max"].isna()

pd.crosstab(
    missing_ni,
    isna
)

max,False,True
ni_centro,,
False,1665,0
True,0,32


In [96]:
missing_ni = df["ni_centro"].isna()
isna = df["pobtotal"].isna()

pd.crosstab(
    missing_ni,
    isna
)

pobtotal,False,True
ni_centro,,
False,1665,0
True,0,32


### `ni_centro` — conclusion

- The field contains 1665 non-null values and 32 missing values.
- All non-null values are mathematically integers, although the field is stored as `float64` because it contains missing values.
- There are 1645 distinct non-null `ni_centro` values.
- Each `ni_centro` is associated with exactly one `nombre`.
- A `ni_centro` may appear in several rows when the same centre has multiple sectors or service types (`tipo_centro`).
- Therefore, `ni_centro` identifies the centre rather than an individual row in the dataset.
- The same 32 records that have no `ni_centro` also have missing `max` and `pobtotal` values and `clasificacion = "SIN RESOLUCIÓN"`.
- `ni_centro` is therefore a strong candidate for `Entidad.cod_entidad` in the global schema.
- However, a strategy will be required for the 32 records where the centre code is missing.

### `ni_centro` — conclusión

- El campo contiene 1665 valores no nulos y 32 valores ausentes.
- Todos los valores no nulos son matemáticamente enteros, aunque el campo está almacenado como `float64` debido a la presencia de valores ausentes.
- Hay 1645 valores distintos de `ni_centro` no nulos.
- Cada `ni_centro` está asociado exactamente a un único `nombre`.
- Un mismo `ni_centro` puede aparecer en varias filas cuando el centro tiene varios sectores o tipos de servicio (`tipo_centro`).
- Por tanto, `ni_centro` identifica al centro y no a una fila individual del conjunto de datos.
- Los mismos 32 registros que no tienen `ni_centro` tampoco tienen valores de `max` ni `pobtotal` y presentan `clasificacion = "SIN RESOLUCIÓN"`.
- Por ello, `ni_centro` es un candidato claro para `Entidad.cod_entidad` en la estructura global.
- Sin embargo, será necesario definir una estrategia para los 32 registros que no disponen de código de centro.

In [98]:
df["id"].dtype

dtype('int64')

In [99]:
(df["id"].dropna() % 1 == 0).all()

np.True_

In [100]:
df["id"].isna().value_counts()

id
False    1697
Name: count, dtype: int64

In [101]:
df["id"].nunique()

1697

In [102]:
df["id"].head(15)

0      1
1      2
2      3
3      4
4      5
5      6
6      7
7      8
8      9
9     10
10    11
11    12
12    13
13    14
14    15
Name: id, dtype: int64

### `id` — conclusion

- The field is stored as `int64`.
- It is complete: all 1697 records contain a value.
- All 1697 values are unique.
- The values form a sequential record identifier.
- The field identifies individual rows in the source dataset rather than physical centres.
- It should therefore not be mapped to `Entidad.cod_entidad`; `ni_centro` is more appropriate for that purpose.

### `id` — conclusión

- El campo está almacenado como `int64`.
- Está completo: los 1697 registros contienen un valor.
- Los 1697 valores son únicos.
- Los valores forman un identificador secuencial de los registros.
- El campo identifica filas individuales del conjunto de datos de origen y no centros físicos.
- Por tanto, no debería asociarse con `Entidad.cod_entidad`; `ni_centro` es más adecuado para ese propósito.

In [103]:
df["WKT"].dtype


dtype('O')

In [104]:
df["WKT"].head(15)


0     POINT (698055 4211247)
1     POINT (694195 4205729)
2     POINT (717780 4247138)
3     POINT (726984 4373714)
4     POINT (700887 4236354)
5     POINT (733612 4321830)
6     POINT (733612 4321830)
7     POINT (726167 4372554)
8     POINT (749729 4270245)
9     POINT (751103 4444477)
10    POINT (756059 4301124)
11    POINT (702873 4231974)
12    POINT (751810 4296427)
13    POINT (751105 4296094)
14    POINT (749634 4295534)
Name: WKT, dtype: object

In [105]:
df["WKT"].isna().value_counts()


WKT
False    1697
Name: count, dtype: int64

In [106]:
df[["WKT", "x_25830", "y_25830"]].head(10)

,WKT,x_25830,y_25830
0,POINT (698055 4211247),698055.0,4211247.0
1,POINT (694195 4205729),694195.0,4205729.0
2,POINT (717780 4247138),717780.0,4247138.0
3,POINT (726984 4373714),726984.0,4373714.0
4,POINT (700887 4236354),700887.0,4236354.0
5,POINT (733612 4321830),733612.0,4321830.0
6,POINT (733612 4321830),733612.0,4321830.0
7,POINT (726167 4372554),726167.0,4372554.0
8,POINT (749729 4270245),749729.0,4270245.0
9,POINT (751103 4444477),751103.0,4444477.0


In [107]:
coords_wkt = df["WKT"].str.extract(
    r"POINT \((-?\d+(?:\.\d+)?) (-?\d+(?:\.\d+)?)\)"
)

coords_wkt.head()

,0,1
0,698055,4211247
1,694195,4205729
2,717780,4247138
3,726984,4373714
4,700887,4236354


In [108]:
coords_wkt.columns = ["wkt_x", "wkt_y"]

coords_wkt = coords_wkt.astype(float)

In [109]:
same_x = coords_wkt["wkt_x"] == df["x_25830"]
same_y = coords_wkt["wkt_y"] == df["y_25830"]

In [110]:
same_coordinates = same_x & same_y

same_coordinates.value_counts()

True    1697
Name: count, dtype: int64

In [111]:
df.loc[
    ~same_coordinates,
    ["WKT", "x_25830", "y_25830"]
]

,WKT,x_25830,y_25830


### `WKT`, `x_25830`, `y_25830` — conclusion

- `WKT` contains point geometries in the form `POINT (X Y)`.
- The coordinates stored in `WKT` match `x_25830` and `y_25830` exactly in all 1697 records.
- Therefore, `WKT` does not provide additional coordinate information; it is another representation of the same point geometry.
- The column names indicate that `x_25830` and `y_25830` use the EPSG:25830 projected coordinate reference system.
- These values are projected X/Y coordinates and must not be used directly as longitude and latitude.
- To populate `Entidad.longitud` and `Entidad.latitud`, the coordinates should be transformed from EPSG:25830 to the geographic coordinate system adopted by the global schema (typically WGS84 / EPSG:4326).
- `WKT` does not require a separate mapping in the current global schema.

### `WKT`, `x_25830`, `y_25830` — conclusión

- `WKT` contiene geometrías de tipo punto con el formato `POINT (X Y)`.
- Las coordenadas almacenadas en `WKT` coinciden exactamente con `x_25830` y `y_25830` en los 1697 registros.
- Por tanto, `WKT` no aporta información adicional sobre las coordenadas, sino que representa la misma geometría de otra forma.
- Los nombres de las columnas indican que `x_25830` y `y_25830` utilizan el sistema de referencia proyectado EPSG:25830.
- Estos valores son coordenadas X/Y proyectadas y no deben utilizarse directamente como longitud y latitud.
- Para obtener `Entidad.longitud` y `Entidad.latitud`, las coordenadas deben transformarse desde EPSG:25830 al sistema geográfico utilizado por la estructura global (normalmente WGS84 / EPSG:4326).
- `WKT` no necesita una correspondencia independiente en la estructura global actual.